### PR TITLE
test(server): optimize multihop worker helper func

### PR DESCRIPTION
**Pull request by ddebko**
# Summary

Replacing the Sleep duration call and utilizing more accurate methods of validating when a worker-controller connection is established for the helper method `NewTestMultihopWorkers`. This should help speed up some of the tests.

### CODE REVIEW

File: internal/daemon/worker/testing.go [hunk 1]
```diff
@@ -552,13 +537,59 @@ func NewTestMultihopWorkers(t testing.TB,
...
+	wg.Wait()
+
+	// validate connectivity between worker pool
+	require.Eventually(func() bool {
+		pkiPath, err := c.DownstreamWorkers().PathTo(c.Context(), pkiWorker.Worker().LastStatusSuccess().WorkerId)
+		if err != nil {
+			return false
+		}
+		childPkiPath, err := c.DownstreamWorkers().PathTo(c.Context(), childPkiWorker.Worker().LastStatusSuccess().WorkerId)
+		if err != nil {
```
Comment (johanbrandhorst) [inline internal/daemon/worker/testing.go:588]: Maybe this should go in a testing_ent instead if we can't do this in community 🤔

Comment (ddebko) [reply internal/daemon/worker/testing.go:588]: Should we move all the tests that reference this helper function into `_ent` files too?

Comment (johanbrandhorst) [reply internal/daemon/worker/testing.go:588]: Yeah that'd be a requirement too.

**Review by johanbrandhorst — APPROVED** [review]
It's a shame to move these tests out of community, but it'll be worth it if the tests are more stable, IMO

**Pull request merged**